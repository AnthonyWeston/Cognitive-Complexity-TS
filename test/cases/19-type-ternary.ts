type E<T> = T extends true ? "true" : "false"; // conditional

// nested conditionals
type F<T> = T extends boolean
    ? T extends true // +1 inherent +0 depth
    // depth 1
        ? // +1 inherent +1 depth
            "true" | "t" // +1 inherent
            : "false"
    : "undefined";
