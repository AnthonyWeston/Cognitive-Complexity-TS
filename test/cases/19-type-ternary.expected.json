{
    "19-type-ternary.ts": {
        "score": 5,
        "inner": [
            {
                "name": "E",
                "score": 1,
                "line": 1,
                "column": 1,
                "inner": [{
                    "name": "T",
                    "score": 0,
                    "line": 1,
                    "column": 8
                }]
            },
            {
                "name": "F",
                "score": 4,
                "line": 4,
                "column": 1,
                "inner": [
                    {
                        "name": "T",
                        "score": 0,
                        "line": 4,
                        "column": 8
                    }
                ]
            }
        ]
    }
}